From 521253a503236395635a2373c1f10a7b07261a19 Mon Sep 17 00:00:00 2001
From: Aikar <aikar@aikar.co>
Date: Tue, 12 May 2015 01:29:17 -0400
Subject: [PATCH] Fix Monument Spawning Behavior

Was spawning in the wrong chunks due to bad getBiome call which uses LATEST worldgen and not the worlds actual biome
---
 src/main/java/net/minecraft/server/WorldGenMonument.java | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/main/java/net/minecraft/server/WorldGenMonument.java b/src/main/java/net/minecraft/server/WorldGenMonument.java
index 7f2738c..23ba06e 100644
--- a/src/main/java/net/minecraft/server/WorldGenMonument.java
+++ b/src/main/java/net/minecraft/server/WorldGenMonument.java
@@ -14,7 +14,7 @@ public class WorldGenMonument extends StructureGenerator {
 
     private int f;
     private int g;
-    public static final List<BiomeBase> d = Arrays.asList(new BiomeBase[] { BiomeBase.OCEAN, BiomeBase.DEEP_OCEAN, BiomeBase.RIVER, BiomeBase.FROZEN_OCEAN, BiomeBase.FROZEN_RIVER});
+    public static final List<BiomeBase> d = Arrays.asList(new BiomeBase[] { BiomeBase.OCEAN, BiomeBase.DEEP_OCEAN}); // EMC
     private static final List<BiomeBase.BiomeMeta> h = Lists.newArrayList();
 
     public WorldGenMonument() {
@@ -63,7 +63,7 @@ public class WorldGenMonument extends StructureGenerator {
         i1 += (random.nextInt(this.f - this.g) + random.nextInt(this.f - this.g)) / 2;
         j1 += (random.nextInt(this.f - this.g) + random.nextInt(this.f - this.g)) / 2;
         if (k == i1 && l == j1) {
-            if (this.c.getWorldChunkManager().getBiome(new BlockPosition(k * 16 + 8, 64, l * 16 + 8), (BiomeBase) null) != BiomeBase.DEEP_OCEAN) {
+            if (!this.c.getWorldChunkManager().a(k * 16 + 8, l * 16 + 8, 7, Arrays.asList(BiomeBase.DEEP_OCEAN))) { // EMC
                 return false;
             }
 
-- 
1.9.1

